sqlglot/parser.py

```diff
@@ -1619,19 +1631,27 @@ def _parse_partitioned_by(self) -> exp.PartitionedByProperty:
             if expr:
                 main_part = expr
             else:
-                main_part = ""
+                main_part = None
 
             if self._match_texts("PARTITIONS"):
-                count_partitions = str(self._parse_schema() or self._parse_bracket(self._parse_field()))
+                count_partitions = str(
+                    self._parse_schema() or self._parse_bracket(self._parse_field())
+                )
 
-        if self._match_texts("HASH"):
-            type="HASH"
+        elif self._match_texts("HASH"):
```

Comment:
Can this be written as:  elif self._match(TokenType.HASH):   ?

Reply:
I'll have to add Token for HASH, but I surely can do that!